src/hooks/useStockWebSocket.ts

```
@@ -0,0 +1,286 @@
+
+      // 자동 재연결 (지수 백오프)
+      if (reconnectAttempts.current < maxReconnectAttempts) {
+        reconnectAttempts.current++;
+        const delay = Math.min(
+          1000 * Math.pow(2, reconnectAttempts.current),
+          30000,
+        );
+        console.log(
+          `[WS] ${delay}ms 후 재연결 시도 (${reconnectAttempts.current}/${maxReconnectAttempts})`,
+        );
+        setTimeout(connect, delay);
```

ESLint

[error] Error: Cannot access variable before it is declared

`connect` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/src/hooks/useStockWebSocket.ts:191:20
  189 |           `[WS] ${delay}ms 후 재연결 시도 (${reconnectAttempts.current}/${maxReconnectAttempts})`,
  190 |         );
> 191 |         setTimeout(connect, delay);
      |                    ^^^^^^^ `connect` accessed before it is declared
  192 |       } else {
  193 |         console.error("[WS] 최대 재연결 시도 횟수 초과");
  194 |       }

/home/jailuser/git/src/hooks/useStockWebSocket.ts:92:3
   90 |   }, []);
   91 |
>  92 |   const connect = useCallback(() => {
      |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>  93 |     if (ws.current?.readyState === WebSocket.OPEN) {
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>  94 |       return;
      …
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 199 |     };
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 200 |   }, [getWsUrl, maxReconnectAttempts, onMessage, onConnectionChange]);
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `connect` is declared here
  201 |
  202 |   // 연결 해제
  203 |   const disconnect = useCallback(() => {

(react-hooks/immutability)